### DISCUSSION TITLE
Update of embedded documents/lists/maps

### DISCUSSION THREAD

**Discussion by mik1234mc** [ArcadeData/arcadedb#742] [Q&A] [answered]
Hi all,

I have a vertex with embedded document and I would like to add fields to this embedded document. Is it possible to do it without replacing the whole embedded document? Same question on list and map, I cant find any relevant example on this. UPDATE SET|MERGE always rewrites the whole embedded document (e.g., `update elttest merge {'c':{'g':'e'}} where @rid = #82:2` and `update elttest merge {'c':{"eee":"rr"}} where @rid = #82:2` results in `{'c':{"eee":"rr"}` not `{'c':{'g':'e',"eee":"rr"}` ).

Thanks,
Michael

**Comment by lvca** [comment, marked as answer]
Try this:

```sql
update elttest set c.g = 'e' where @rid = #82:2
```

**Reply by mik1234mc** [reply]
Thanks, this works
